EIPS/eip-00.md

```diff
@@ -0,0 +1,828 @@
+
+| Opcode Name | Solidity Equivalent | Value                                                                         |
+|-------------|---------------------|-------------------------------------------------------------------------------|
+| `CALLER`    | `msg.sender`        | The `AA_ENTRY_POINT` address. `AA_SENDER_CREATOR` for the "deployment frame". |
+| `ORIGIN`    | `tx.origin`         | The transaction `sender` address                                              |
+| `CALLDATA*` | `msg.data`          | The transaction data is set to inputs of the corresponding frame              |
+
+#### Nonce validation frame
+
+The `NonceManager` is invoked with the following data:
+
+```solidity
+abi.encodePacked(sender, nonce)
+```
+
+#### Sender deployment frame
+
+The `deployer` address is invoked with the `deployerData[20:]` as call data input.
+It is important that the `deployer` is **not** invoked from the `AA_ENTRY_POINT` but from the `AA_SENDER_CREATOR`.
+This is necessary to guarantee that `AA_ENTRY_POINT` may never initiate a call to a `sender` execution function
+without first completing a successful validation.
+
+The gas limit of this frame is set to `validationGasLimit`.
+The amount of gas used by this frame is referred to as `senderCreationGasUsed`.
+
+The sender deployment frame MUST result in the `sender` address becoming
+initialized with contract code.
+
+#### Sender validation frame
+
+We define the following Solidity struct to represent the AA transaction on-chain:
+
+```solidity
+
+struct TransactionType4 {
+    address sender;
+    uint256 nonce;
+    uint256 validationGasLimit;
+    uint256 paymasterGasLimit;
+    uint256 callGasLimit;
+    uint256 maxFeePerGas;
+    uint256 maxPriorityFeePerGas;
+    uint256 builderFee;
+    bytes paymasterData;
+    bytes deployerData;
+    bytes callData;
+    bytes signature;
+}
+
+```
+
+We then define the following Solidity method and the `sender` of the transaction is invoked with the corresponding data:
+
+```solidity
+
+function validateTransaction(uint256 version, bytes32 txHash, bytes transaction) external returns (uint256 validationData);
+
+```
+
+The gas limit of this frame is set to `validationGasLimit - senderCreationGasUsed`.\
+The `transaction` parameter is interpreted as an ABI encoding of `TransactionType4`.\
+The `txHash` parameter represents the hash of the AA_TX_TYPE transaction with empty signature, as defined in section
+[Calculation of Transaction Type AA_TX_TYPE hash](#calculation-of-transaction-type-aatxtype-hash).\
+The `version` parameter is added in order to maintain the Solidity method ID in case of changes to this struct
+in future revisions of this EIP.
+
+The amount of gas used by this frame is referred to as `senderValidationGasUsed`.
+
+The frame must return 32 bytes `validationData` that is interpreted as:
+
+```solidity
+
+abi.encodePacked(MAGIC_VALUE_SENDER, validBefore, validAfter)
```

Comment:
The type and the usage of `validBefore` and `validAfter` should be defined somewhere.

Reply:
@forshtat looks like we missed defining the types.  Also, we used `validBefore` and `validUntil` interchangeably.  In 4337 there is only `validUntil` so let's just call it that.

@juniset the usage is described in the "Transaction validity time range parameters" section (line 521) and shown in the pseudocode (line 442,426,400), but we should add the type definition.